test/e2e/reconciler/apiexportendpointslice/apiexportendpointslice_test.go

```diff
@@ -0,0 +1,309 @@
+	require.NoError(t, err)
+
+	require.Eventually(t, func() bool {
+		slice, err = kcpClusterClient.Cluster(partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Get(ctx, slice.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+		return conditions.IsTrue(slice, apisv1alpha1.APIExportValid)
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected valid APIExport")
+	require.True(t, conditions.IsTrue(slice, apisv1alpha1.APIExportEndpointSliceURLsReady), "expecting URLs ready condition")
+
+	t.Logf("Creating the Partition")
+	partitionClient := kcpClusterClient.TopologyV1alpha1().Partitions()
+	_, err = partitionClient.Cluster(partitionClusterPath).Create(ctx, partition, metav1.CreateOptions{})
+	require.NoError(t, err, "error creating Partition")
+
+	t.Logf("Adding a Partition to the APIExportEndpointSlice")
+	slice.Spec.Partition = partition.Name
+	_, err = kcpClusterClient.Cluster(partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Update(ctx, slice, metav1.UpdateOptions{})
+	require.NoError(t, err, "error updating APIExportEndpointSlice")
+
+	require.Eventually(t, func() bool {
+		s, err := kcpClusterClient.Cluster(partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Get(ctx, slice.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+		return conditions.IsTrue(s, apisv1alpha1.PartitionValid)
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected valid Partition")
+	require.True(t, conditions.IsTrue(slice, apisv1alpha1.APIExportEndpointSliceURLsReady), "expecting URLs ready condition")
+
+	t.Logf("Checking that no endpoint has been populated")
+	require.Eventually(t, func() bool {
+		s, err := kcpClusterClient.Cluster(partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Get(ctx, slice.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+		return len(s.Status.APIExportEndpoints) == 0
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "not expecting any endpoint")
+
+	// Endpoint tests require the edition of shards.
+	// These tests are run on a private cluster to avoid side effects on other e2e tests.
+	// They require the resources previously created: APIExport, APIExportEndpointSlice, etc.
+	shard := &corev1alpha1.Shard{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "my-shard",
+			Labels: map[string]string{
+				"region": "apiexportendpointslice-test-region",
+			},
+		},
+		Spec: corev1alpha1.ShardSpec{
+			BaseURL: "https://base.kcp.test.dev",
+		},
+	}
+
+	t.Logf("Creating a shard in the region")
+	shardClient := kcpClusterClient.CoreV1alpha1().Shards()
+	shard, err = shardClient.Cluster(core.RootCluster.Path()).Create(ctx, shard, metav1.CreateOptions{})
+	require.NoError(t, err, "error creating Shard")
+
+	require.Eventually(t, func() bool {
+		s, err := kcpClusterClient.Cluster(partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Get(ctx, slice.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+		return len(s.Status.APIExportEndpoints) == 1
```

Comment:
shouldn't we check the content of `s.Status.APIExportEndpoints` it should match `shard.Spec.VirtualWorkspaceURL` ?

Reply:
added